### PR TITLE
[WIP] Add multiple domains feature in request

### PR DESCRIPTION
**Pull request by tgalopin**

### CODE REVIEW

File: src/Cli/Command/Helper/DistinguishedNameHelper.php [hunk 1]
```diff
@@ -82,6 +86,31 @@ public function ask(QuestionHelper $helper, InputInterface $input, OutputInterfa
             'What is your e-mail address (field "E" of the distinguished name)? : '
         ));
 
+        $alternateDomains = [];
+
+        if (!$disableAlternateDomains) {
+            $output->write("\n");
+            $output->writeln('Do you want to handle multiple domains with this certificate?');
+            $output->writeln('Please be aware that this may reduce your certificate compatibility with very old devices.');
```
Comment (jderusse) [inline src/Cli/Command/Helper/DistinguishedNameHelper.php]: Not anymore, with the new X3 intermediate certificate

File: src/Cli/Command/Helper/DistinguishedNameHelper.php [hunk 2]
```diff
@@ -82,6 +86,31 @@ public function ask(QuestionHelper $helper, InputInterface $input, OutputInterfa
...
+            $output->writeln('Full compatibility table is available at https://en.wikipedia.org/wiki/Server_Name_Indication#Support');
+            $output->write("\n");
+
+            $sni = $helper->ask($input, $output, new ConfirmationQuestion(
```
Comment (jderusse) [inline src/Cli/Command/Helper/DistinguishedNameHelper.php]: SANs are not related to SNI

File: src/Cli/Command/RequestCommand.php [hunk 3]
```diff
@@ -59,6 +59,7 @@ protected function configure()
                 new InputOption('organization', null, InputOption::VALUE_REQUIRED, 'Your organization/company (field "O" of the distinguished name, for instance: "Acme PHP")'),
                 new InputOption('unit', null, InputOption::VALUE_REQUIRED, 'Your unit/department in your organization (field "OU" of the distinguished name, for instance: "Sales")'),
                 new InputOption('email', null, InputOption::VALUE_REQUIRED, 'Your e-mail address (field "E" of the distinguished name)'),
+                new InputOption('alternate-domains', null, InputOption::VALUE_REQUIRED, 'Alternate domains for this certificate, separated by commas (use "none" to disable the question in the command)'),
```
Comment (jderusse) [inline src/Cli/Command/RequestCommand.php]: What's about using the `InputOption::VALUE_IS_ARRAY`?

File: src/Cli/Command/Helper/DistinguishedNameHelper.php [hunk 4]
```diff
@@ -33,29 +34,32 @@ public function getName()
 
     /**
      * @param DistinguishedName $distinguishedName
+     * @param bool $disableAlternateDomains
```
Comment (jderusse) [inline src/Cli/Command/Helper/DistinguishedNameHelper.php]: bad alignment

File: src/Cli/Command/Helper/DistinguishedNameHelper.php [hunk 5]
```diff
@@ -33,29 +34,32 @@ public function getName()
...
     {
         return $distinguishedName->getCountryName()
             && $distinguishedName->getStateOrProvinceName()
             && $distinguishedName->getLocalityName()
             && $distinguishedName->getOrganizationName()
             && $distinguishedName->getOrganizationalUnitName()
             && $distinguishedName->getEmailAddress()
-            && $distinguishedName->getCommonName();
+            && $distinguishedName->getCommonName()
+            && $disableAlternateDomains;
```
Comment (jderusse) [inline src/Cli/Command/Helper/DistinguishedNameHelper.php]: That's mean, certificate with alternateSubjectName will never be ready?
What's about ignoring SANs and automatically upgrading the DistinguishedName ? (see my PR)

**Comment by tgalopin** [comment]
Fixed by https://github.com/acmephp/acmephp/pull/24

**Pull request closed without merging**